.orchestra/config/components/aws-sdk.yml

```diff
@@ -0,0 +1,78 @@
+#@ load("@ytt:overlay", "overlay")
+#@ load("/lib/ninja.lib.yml", "ninja")
+#@ load("/lib/create-component.lib.yml", "single_build_component")
+
+#@yaml/text-templated-strings
+---
+#@ def aws_sdk_args():
+#! version 1.11.113
+#@ target_sha1 = "ece7e32dbca87b3fbc7edf0e81d00f328f4b2b07"
+#@ source_url = "https://github.com/aws/aws-sdk-cpp"
+license: source/LICENSE
+configure: |
+  mkdir -p "$BUILD_DIR"
+  cd "$BUILD_DIR"
+
+  if [[ -f .last-hash && "$(cat .last-hash)" != '(@= target_sha1 @)' ]]; then
+    # Hash changed
+    rm -rf -- source
+  fi
+
+  if [[ ! -f .last-hash && -d source ]]; then
+    # Dirty checkout
+    rm -rf -- source
+  fi
+
+  if [ ! -d source ]; then
+    mkdir source
+    pushd source &> /dev/null
+
+    git init --quiet
+    git remote add origin (@= source_url @)
+    git pull --quiet --ff-only --depth=1 origin (@= target_sha1 @)
+    git submodule update --quiet --init --recursive --depth=1
+
+    patch -p1 < "${ORCHESTRA_DOTDIR}/patches/aws-sdk-force-shared.patch"
+
+    popd &> /dev/null
+    echo '(@= target_sha1 @)' > .last-hash
+  fi
+
+  if [ -d build ]; then
+    rm -rf -- build
+  fi
+  mkdir build
+  cd build
```

Comment:
See what `sqlite.yml` does in terms of organizing the build directory. Let's stick with current practices.